packages/@aws-cdk/aws-s3-deployment/lambda/src/index.py

```diff
@@ -0,0 +1,120 @@
+import subprocess
+import os
+import tempfile
+import json
+import json
+import traceback
+import logging
+from botocore.vendored import requests
+from zipfile import ZipFile
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+CFN_SUCCESS = "SUCCESS"
+CFN_FAILED = "FAILED"
+
+def handler(event, context):
+
+    def cfn_error(message=None):
+        logger.info("| cfn_error: %s" % message)
+        cfn_send(event, context, CFN_FAILED, reason=message)
+
+    try:
+        logger.info(event)
+
+        # cloudformation request type (create/update/delete)
+        request_type = event['RequestType']
+
+        # extract resource properties
+        props = event['ResourceProperties']
+
+        try:
+            source_bucket_name = props['SourceBucketName']
+            source_object_key  = props['SourceObjectKey']
+            dest_bucket_name   = props['DestinationBucketName']
+            dest_bucket_prefix = props.get('DestinationBucketKeyPrefix', '')
+            retain_on_delete   = props.get('RetainOnDelete', "false") == "true"
+        except KeyError as e:
+            cfn_error("missing request resource property %s" % str(e))
+            return
+
+        s3_source_zip = "s3://%s/%s" % (source_bucket_name, source_object_key)
+        s3_dest = "s3://%s/%s" % (dest_bucket_name, dest_bucket_prefix)
+        logger.info("| s3_dest: %s" % s3_dest)
+
+        # delete or create/update
+        if request_type == "Delete":
```

Comment:
Aaaaactually you probably also need to do this on `Update`; it might be that your resource got pointed to a different bucket, and so you probably want to delete from the old one (`OldResourceProperties`)).

Sorry for not thinking of this earlier.

Reply:
Good catch! 

I guess the safest thing to do is if the destination bucket+prefix is different from the new bucket+prefix, we first delete the old one and then sync to the new one. Makes sense?